tsec-libsodium/src/main/scala/tsec/libsodium/kdf/KeyDerivation.scala

```diff
@@ -0,0 +1,37 @@
+package tsec.libsodium.kdf
+
+import cats.data.StateT
+import cats.effect.Sync
+import tsec.libsodium.ScalaSodium
+import tsec.libsodium.cipher.SodiumKey
+import tsec.libsodium.pk.PrivateKey
+
+sealed trait KeyDerivation
+
+object KeyDerivation {
+
+  def generateKey[F[_]](implicit F: Sync[F], S: ScalaSodium): F[PrivateKey[KeyDerivation]] = F.delay {
+    val masterKey = PrivateKey[KeyDerivation](new Array[Byte](ScalaSodium.crypto_kdf_KEYBYTES))
+    S.crypto_kdf_keygen(masterKey)
+    masterKey
+  }
+
+  def deriveKey[F[_]](
+      masterKey: PrivateKey[KeyDerivation],
+      keyLength: Int,
+      context: Array[Byte]
+  )(implicit F: Sync[F], S: ScalaSodium): StateT[F, Int, SodiumKey[KeyDerivation]] =
```

Comment:
You need to have the ability to get the same id twice. I don't think this helper is necessary.